src/task.c

```diff
@@ -833,17 +845,18 @@ task_system_create(platform_heap_id    hid,
          *system = NULL;
          return rc;
       }
-      nbg_threads_created += (use_bg_threads ? num_bg_threads[type] : 0);
-   }
-
-   // Wait-for all background threads to start-up and register. This is not
-   // really needed right now, but it allows for reliable unit-testing to verify
-   // state of the task system when bg threads are configured.
-   threadid *max_tid = task_system_get_max_tid(ts);
-   while (*max_tid < nbg_threads_created) {
-      platform_sleep(USEC_TO_NSEC(100000)); // 100 msec.
-      max_tid = task_system_get_max_tid(ts);
+      uint64 nbg_threads = num_bg_threads[type];
+      if (nbg_threads) {
+         platform_default_log("Splinter task system created %lu"
+                              " background thread%sof type '%s'.\n",
+                              nbg_threads,
+                              ((nbg_threads > 1) ? "s " : " "),
+                              task_type_name[type]);
+      }
    }
+   debug_assert((*system == NULL),
```

Comment:
This is a minor cleanliness / hygiene check. In my unit-tests, where I am destroying and recreating the task-system, I wanted to be sure that the task system interface is not inadvertently clobbering a valid task_system`ts *`. Hence, this assert.

This cascades into changes in many consumer code paths where on-stack `ts *` had to be initialized to NULL, which is a good thing to do anyway.